api/v1/receiver_types.go

```diff
@@ -63,10 +63,24 @@ type ReceiverSpec struct {
 	// +optional
 	Events []string `json:"events,omitempty"`
 
+	// TODO: Validate one or other (or both?)
+
```

Comment:
This will need to be addressed, because you could have `.spec.resourceExpressions` or `.spec.resources` (or both).